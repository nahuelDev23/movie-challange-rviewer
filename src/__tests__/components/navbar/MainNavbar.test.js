import { shallow } from 'enzyme'
import { mount } from 'enzyme/build';
import { MainNavbar } from '../../../components/navbar/MainNavbar';
import { UiContext } from '../../../context/ui/UiContext';

describe('Pruebas en <MainNavbar/>', () => {
    test('Debe mostrar <MainNavbar/> correctamente', () => {
        const wrapper = shallow(<MainNavbar />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe contener solo una etiqueta nav', () => {
        const wrapper = shallow(<MainNavbar />);
        expect(wrapper.find('nav').length).toBe(1);
    })

    test('Debe haber un link que al hacer click re direccione al home', () => {
        const wrapper = shallow(<MainNavbar />);
        const link = wrapper.find('Link');
        expect(link.at(0).props().to).toBe('/');
        expect(link.at(2).props().to).toBe('#');
    })

    test('Debe haber dos componentes ActiveLink', () => {
        const wrapper = shallow(<MainNavbar />);
        const link = wrapper.find('ActiveLink');
        expect(link.length).toBe(2);
    })

    test('Debe haber un link que al hacer click re direccione a la pagina movies', () => {
        const wrapper = shallow(<MainNavbar />);
        const link = wrapper.find('ActiveLink');
        expect(link.at(0).props().to).toBe('/movies');
    })

    test('Debe haber un link que al hacer click re direccione a la pagina series', () => {
        const wrapper = shallow(<MainNavbar />);
        const link = wrapper.find('ActiveLink');
        expect(link.at(1).props().to).toBe('/series');
    })

    
})