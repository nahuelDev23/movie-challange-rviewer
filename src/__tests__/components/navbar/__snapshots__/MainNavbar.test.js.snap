// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MainNavbar/> Debe mostrar <MainNavbar/> correctamente 1`] = `
<nav
  className="nav center-content"
>
  <div
    className="nav__left"
  >
    <div
      className="nav__logo"
    >
      <Link
        to="/"
      >
         
        <img
          alt="Logo Dreadful Tomato"
          src="logo.png"
        />
      </Link>
    </div>
    <div
      className="nav__links"
    >
      <ActiveLink
        to="/movies"
      >
        <img
          alt="icon_movies"
          src="icon-movies.png"
        />
        <p>
          Movies
        </p>
      </ActiveLink>
      <ActiveLink
        to="/series"
      >
        <img
          alt="icon_series"
          src="icon-series.png"
        />
        <p>
          Series
        </p>
      </ActiveLink>
    </div>
  </div>
  <div
    className="nav__actions"
  >
    <div
      className="nav__filters undefined "
    >
      <img
        alt="Icon filter"
        src="icon-filter.png"
      />
      <p>
        filters
      </p>
    </div>
    <div
      className="nav__login"
    >
      <p>
        <Link
          to="#"
        >
          Login
        </Link>
      </p>
      <img
        alt="icono para el login"
        src="icon-login.png"
      />
    </div>
    <div
      className="nav__trial btn btn--active"
    >
      <Link
        to="#"
      >
         Start your free trial
      </Link>
    </div>
  </div>
</nav>
`;
